tools/check_version.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+
+import sys
+from typing import List, Callable
+from ctypes import c_uint8, c_uint16
+import argparse
+import re
+
+#
+# MARK: - Standards
+#
+
+class Standard:
+    def __init__(self, is_version_compliant:Callable[[str], bool], success_message:str, error_message:str):
+        self.is_version_compliant:Callable[[str], bool] = is_version_compliant
+        self.success_message:str = success_message
+        self.error_message:str = error_message
```

Comment:
why is this needed?

I'm not a python expert, but I find this very hard to read and I'm not sure I understand the purpose